pkg/odo/cli/component/devfile.go

```diff
@@ -169,6 +169,12 @@ func (do *DeleteOptions) DevfileComponentDelete() error {
 		return err
 	}
 
+	componentExists, _ := devfileHandler.DoesComponentExist(componentName)
+	if !componentExists {
+		log.Italicf("Component %s does not exist", componentName)
+		return nil
+	}
```

Comment:
We are already doing the existence check in the kubernetes adapter https://github.com/openshift/odo/blob/edc7c336973c3978d3e8cadfee99e1c3dff2b91c/pkg/devfile/adapters/kubernetes/component/adapter.go#L445. Also we shouldn't we ignoring this error and should return it.

Reply:
We cant use the err here.
1. we're explicitly passing nil from kube, see comment https://github.com/openshift/odo/pull/3405/files#r446364162
2. docker's `utils.ComponentExists()` is a special case since docker does not have a deployment and has multiple individual containers - see code https://github.com/openshift/odo/blob/master/pkg/devfile/adapters/docker/utils/utils.go#L43. So if for example, len(containers) != len(devfile components), we return an error acc to the code but odo delete should be able to delete those remaining/hanging containers and this is the reason we ignore err here

Reply:
> we're explicitly passing nil from kube, see comment #3405 (comment)

https://github.com/openshift/odo/pull/3405#discussion_r446839977 will solve this problem

> So if for example, len(containers) != len(devfile components), we return an error acc to the code but odo delete should be able to delete those remaining/hanging containers and this is the reason we ignore err here

Then let's move this check from move here into the delete handler. That way it won't affect the kube part and also it will be consistent with the kubernetes delete handler which has the exist check in it https://github.com/openshift/odo/blob/edc7c336973c3978d3e8cadfee99e1c3dff2b91c/pkg/devfile/adapters/kubernetes/component/adapter.go#L445

Reply:
Moved over the logic to the adapter check. This also resulted in moving over the spinners to the individual adapters to keep things clean from UX pov.